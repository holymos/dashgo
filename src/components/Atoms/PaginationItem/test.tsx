import { render, screen } from "@testing-library/react";
import { PaginationItem } from ".";

describe("<PaginationItem />", () => {
  it("should render correctly", () => {
    render(<PaginationItem pageNumber={1} />);

    expect(screen.getByText(/1/i)).toBeInTheDocument();
  });

  it("should render a disabled button with pink background if is active", () => {
    const { container } = render(<PaginationItem pageNumber={1} isCurrent />);

    expect(screen.getByText(/1/i)).toBeDisabled();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render a disabled button with pink background if is active", () => {
    const { container } = render(<PaginationItem pageNumber={1} />);

    expect(screen.getByText(/1/i)).toBeEnabled();
    expect(container.firstChild).toMatchSnapshot();
  });
});
