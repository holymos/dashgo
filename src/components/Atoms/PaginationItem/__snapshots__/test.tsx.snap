// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaginationItem /> should render a disabled button with pink background if is active 1`] = `
<button
  class="pagination-button active"
  disabled=""
>
  1
</button>
`;

exports[`<PaginationItem /> should render a disabled button with pink background if is active 2`] = `
<button
  class="pagination-button"
>
  1
</button>
`;
